install-panel.sh

```diff
@@ -585,139 +631,85 @@ function centos8_dep {
+  [ "$CONFIGURE_LETSENCRYPT" == true ] && ufw allow https > /dev/null
 
   ufw enable
   ufw status numbered | sed '/v6/d'
 }
 
-function firewall_firewalld {
+firewall_firewalld() {
   echo -e "\n* Enabling firewall_cmd (firewalld)"
-  echo "* Opening port 22 (SSH), 80 (HTTP) and 443 (HTTPS)"
-
-  if [ "$OS_VER_MAJOR" == "7" ]; then
-    # pointing to /dev/null silences the command output
-    echo "* Installing firewall"
-    yum -y -q update > /dev/null
-    yum -y -q install firewalld > /dev/null
-
-    systemctl --now enable firewalld > /dev/null # Start and enable
-    firewall-cmd --add-service=http --permanent -q # Port 80
-    firewall-cmd --add-service=https --permanent -q # Port 443
-    firewall-cmd --add-service=ssh --permanent -q  # Port 22
-    firewall-cmd --reload -q # Enable firewall
-
-  elif [ "$OS_VER_MAJOR" == "8" ]; then
-    # pointing to /dev/null silences the command output
-    echo "* Installing firewall"
-    dnf -y -q update > /dev/null
-    dnf -y -q install firewalld > /dev/null
-
-    systemctl --now enable firewalld > /dev/null # Start and enable
-    firewall-cmd --add-service=http --permanent -q # Port 80
-    firewall-cmd --add-service=https --permanent -q # Port 443
-    firewall-cmd --add-service=ssh --permanent -q  # Port 22
-    firewall-cmd --reload -q # Enable firewall
+  echo "* Opening port 22 (SSH), 80 (HTTP) and (optionaly) 443 (HTTPS)"
 
-  else
-    print_error "Unsupported OS"
-    exit 1
-  fi
+  # Install
+  [ "$OS_VER_MAJOR" == "7" ] && yum -y -q install firewalld > /dev/null
+  [ "$OS_VER_MAJOR" == "8" ] && dnf -y -q install firewalld > /dev/null
+
+  # Enable
+  systemctl --now enable firewalld > /dev/null # Enable and start
+
+  # Configure
+  firewall-cmd --add-service=http --permanent -q # Port 80
+  [ "$CONFIGURE_LETSENCRYPT" == true ] && firewall-cmd --add-service=https --permanent -q # Port 443
+  firewall-cmd --add-service=ssh --permanent -q  # Port 22
+  firewall-cmd --reload -q # Enable firewall
 
   echo "* Firewall-cmd installed"
   print_brake 70
 }
 
-function letsencrypt {
+letsencrypt() {
   FAILED=false
 
   # Install certbot
-  if [ "$OS" == "debian" ] || [ "$OS" == "ubuntu" ]; then
-    apt-get install certbot -y
-  elif [ "$OS" == "centos" ]; then
-    [ "$OS_VER_MAJOR" == "7" ] && yum install certbot
-    [ "$OS_VER_MAJOR" == "8" ] && dnf install certbot
-  else
-    # exit
-    print_error "OS not supported."
-    exit 1
-  fi
-
-  # Stop nginx
-  systemctl stop nginx
+  case "$OS" in
+    debian | ubuntu)
+      apt-get -y install certbot python3-certbot-nginx
+      ;;
+    centos)
+      [ "$OS_VER_MAJOR" == "7" ] && yum -y -q install certbot python-certbot-nginx
+      [ "$OS_VER_MAJOR" == "8" ] && dnf -y -q install certbot python3-certbot-nginx
+      ;;
+  esac
 
   # Obtain certificate
-  certbot certonly --no-eff-email --email "$email" --standalone -d "$FQDN" || FAILED=true
+  certbot --nginx --redirect --no-eff-email --email "$email" -d "$FQDN" || FAILED=true
```

Comment:
So as I remember the last time I checked I am assigned to this. It doesn't change that much.

1. The nginx_ssl.conf is removed because using `--nginx` it creates ssl config from the non ssl config.
2. Assuming ssl isn't that straight forward. I left all the lines for getting user input, but it doesn't work.

Problem is talked about in:

> So basically there is no way to assume_ssl after using certbot with --nginx, because the files aren't edited if certificates fail. I don't see this as a flaw.

Now that I think I should add assuming ssl back, but in a bit different form. It would only be available in the start and if certificates fail, users would have the non-ssl config.
Or another option might be re downloading the nginx config files and editing them the second time, but with assume ssl.

Reply:
> So as I remember the last time I checked I am assigned to this. It doesn't change that much.
> 
> 1. The nginx_ssl.conf is removed because using `--nginx` it creates ssl config from the non ssl config.
> 2. Assuming ssl isn't that straight forward. I left all the lines for getting user input, but it doesn't work.
> 
> Problem is talked about in:
> 
> > So basically there is no way to assume_ssl after using certbot with --nginx, because the files aren't edited if certificates fail. I don't see this as a flaw.
> 
> Now that I think I should add assuming ssl back, but in a bit different form. It would only be available in the start and if certificates fail, users would have the non-ssl config.
> Or another option might be re downloading the nginx config files and editing them the second time, but with assume ssl.

I assigned it to you after you submitted the PR, not before. Not too sure what you mean here. If people want to assume SSL but not use Let's Encrypt, the `nginx_ssl.conf` file perhaps is best to be kept but not downloaded when using Nginx; only when using `assume_ssl`?

Reply:
Yeah that's pretty easy, but the hard thing is assuming ssl after a failed letsencrypt.. It could be done, and I think I have a solution in my head.

I can easily add back assume ssl if user chooses out right, and at the same time think up the solution to assuming ssl after failed letsencrypt.

Reply:
Ok I think I have added back assuming ssl.